packages/backend/server/src/models/comment.ts

```diff
@@ -0,0 +1,336 @@
+  // #region Comment
+
+  /**
+   * Create a comment
+   * @param input - The comment create input
+   * @returns The created comment
+   */
+  async create(input: CommentCreate) {
+    const data = CommentCreateSchema.parse(input);
+    return await this.db.comment.create({
+      data,
+    });
+  }
+
+  async get(id: string) {
+    return await this.db.comment.findUnique({
+      where: { id, deletedAt: null },
+    });
+  }
+
+  /**
+   * Update a comment content
+   * @param input - The comment update input
+   * @returns The updated comment
+   */
+  async update(input: CommentUpdate) {
+    const data = CommentUpdateSchema.parse(input);
+    return await this.db.comment.update({
+      where: { id: data.id },
+      data: {
+        content: data.content,
+      },
+    });
+  }
+
+  /**
+   * Delete a comment or reply
+   * @param id - The id of the comment or reply
+   * @returns The deleted comment or reply
+   */
+  async delete(id: string) {
+    await this.db.comment.update({
+      where: { id },
+      data: { deletedAt: new Date() },
+    });
+    this.logger.log(`Comment ${id} deleted`);
+  }
+
+  /**
+   * Resolve a comment or not
+   * @param input - The comment resolve input
+   * @returns The resolved comment
+   */
+  async resolve(input: CommentResolve) {
+    const data = CommentResolveSchema.parse(input);
+    return await this.db.comment.update({
+      where: { id: data.id },
+      data: { resolved: data.resolved },
+    });
+  }
+
+  async count(workspaceId: string, docId: string) {
+    return await this.db.comment.count({
+      where: { workspaceId, docId, deletedAt: null },
+    });
+  }
+
+  /**
+   * List comments ordered by sid descending
+   * @param workspaceId - The workspace id
+   * @param docId - The doc id
+   * @param options - The options
+   * @returns The list of comments with replies
+   */
+  async list(
+    workspaceId: string,
+    docId: string,
+    options?: {
+      sid?: number;
+      take?: number;
+    }
+  ): Promise<CommentWithReplies[]> {
+    const comments = (await this.db.comment.findMany({
+      where: {
+        workspaceId,
+        docId,
+        ...(options?.sid ? { sid: { lt: options.sid } } : {}),
+        deletedAt: null,
+      },
+      orderBy: { sid: 'desc' },
+      take: options?.take ?? 100,
+    })) as Comment[];
+
+    const replies = (await this.db.reply.findMany({
+      where: {
+        commentId: { in: comments.map(comment => comment.id) },
+        deletedAt: null,
+      },
+      orderBy: { sid: 'asc' },
+    })) as Reply[];
+
+    const replyMap = new Map<string, Reply[]>();
+    for (const reply of replies) {
+      const items = replyMap.get(reply.commentId) ?? [];
+      items.push(reply);
+      replyMap.set(reply.commentId, items);
+    }
+
+    const commentWithReplies = comments.map(comment => ({
+      ...comment,
+      replies: replyMap.get(comment.id) ?? [],
+    }));
+
+    return commentWithReplies;
+  }
+
+  async listChanges(
```

Comment:
what's this for?

Reply:
After the initial loading is complete, periodic polling is required to obtain subsequent changes in data

See https://insider.affine.pro/workspace/af3478a2-9c9c-4d16-864d-bffa1eb10eb6/fKah3bNZNY?mode=page&blockIds=IFvGsQ6i15